projects/igniteui-angular/src/lib/grids/grid-base.directive.ts

```diff
@@ -1392,8 +1392,9 @@ export abstract class IgxGridBaseDirective extends DisplayDensityBase implements
         }
         this.advancedFilteringExpressionsTreeChange.emit(this._advancedFilteringExpressionsTree);
 
-        if (this.filteringService.isFilteringExpressionsTreeEmpty(this._advancedFilteringExpressionsTree) &&
-            !this.advancedFilteringExpressionsTree) {
+        if (this.filteringService.isFilteringExpressionsTreeEmpty(this._filteringExpressionsTree) &&
+            (this.filteringService.isFilteringExpressionsTreeEmpty(this.advancedFilteringExpressionsTree)
```

Comment:
`this.filteringService.isFilteringExpressionsTreeEmpty(this.advancedFilteringExpressionsTree` wil lreturn true if `advancedFilteringExpressionsTree` is `null` or `undefined`. So no need of the second NOT operator.

Reply:
Did this in my new commit 👍